.github/workflows/misc.yml

```diff
@@ -0,0 +1,120 @@
+name: XGBoost CI (misc)
+
+on: [push, pull_request]
+
+permissions:
+  contents: read  # to fetch code (actions/checkout)
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  BRANCH_NAME: >-
+    ${{ github.event.pull_request.number && 'PR-' }}${{ github.event.pull_request.number || github.ref_name }}
+
+jobs:
+  gtest-cpu:
+    name: Test Google C++ test (CPU)
+    runs-on: macos-13
+    steps:
+      - uses: actions/checkout@v4.2.2
+        with:
+          submodules: 'true'
+      - name: Install system packages
+        run: |
+          brew install ninja libomp
+      - name: Build gtest binary
+        run: |
+          mkdir build
+          cd build
+          cmake .. -DGOOGLE_TEST=ON -DUSE_OPENMP=ON -DUSE_DMLC_GTEST=ON -GNinja -DBUILD_DEPRECATED_CLI=ON -DUSE_SANITIZER=ON -DENABLED_SANITIZERS=address -DCMAKE_BUILD_TYPE=RelWithDebInfo
+          ninja -v
+      - name: Run gtest binary
+        run: |
+          cd build
+          ./testxgboost
+          ctest -R TestXGBoostCLI --extra-verbose
+
+  gtest-cpu-nonomp:
+    name: Test Google C++ unittest (CPU Non-OMP)
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4.2.2
+        with:
+          submodules: 'true'
+      - name: Install system packages
+        run: |
+          sudo apt-get install -y --no-install-recommends ninja-build
+      - name: Build and install XGBoost
+        shell: bash -l {0}
+        run: |
+          mkdir build
+          cd build
+          cmake .. -GNinja -DGOOGLE_TEST=ON  -DUSE_DMLC_GTEST=ON -DUSE_OPENMP=OFF -DBUILD_DEPRECATED_CLI=ON
+          ninja -v
+      - name: Run gtest binary
+        run: |
+          cd build
+          ctest --extra-verbose
+
+  c-api-demo:
+    name: Test installing XGBoost lib + building the C API demo
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+      - uses: actions/checkout@v4.2.2
+        with:
+          submodules: 'true'
+      - uses: conda-incubator/setup-miniconda@v3.1.0
```

Comment:
This gets reused enough that I think it is worth having your own [local custom action](https://docs.github.com/en/actions/sharing-automations/creating-actions/creating-a-composite-action). It would change your code here to be something like:

```
uses: dmlc/xgboost/actions/miniforge-setup@main
```

where `actions/miniforge-setup/action.yml` would look like:
```
name: 'Setup miniforge'
inputs:
  environment_file: # this seems like the one parameter you might need to always pass in
    description: "environment file!!!"
    required: true
  <Optionally accept parameters to pass through to miniconda action, but make defaults be workable>
runs:
  using: "composite"
  steps:
   - uses: conda-incubator/setup-miniconda@v3.1.0
     with:
          miniforge-variant: Miniforge3
          miniforge-version: latest
          activate-environment: cpp_test
          environment-file: ops/conda_env/cpp_test.yml
          use-mamba: true
```

"dispatch scripts" might be helpful to decouple the version of the actions from the version of the repo. I described these in https://github.com/rapidsai/shared-actions